src/commands/cmd_json.cc

```diff
@@ -52,7 +52,33 @@ class CommandJsonGet : public Commander {
   }
 };
 
+class CommandJsonStrAppend : public Commander {
+public:
+    Status Execute(Server *svr, Connection *conn, std::string *output) override {
+      redis::Json json(svr->storage, conn->GetNamespace());
+
+      JsonValue result;
+      std::vector<std::string> paths;
+      std::vector<std::string> values;
+      for (int i=2;i<(int)args_.size();++i) {
+        auto val = args_[i];
+        if (val.substr(0, 1) == "$") {
+          paths.push_back(val);
+        } else{
+          values.push_back(val);
+        }
+      }
+      uint64_t cnt = 0;
+      auto s = json.StrAppend(args_[1], {args_.begin() + 2, args_.end() - 1}, args_.back(), cnt);
+      if (!s.ok()) return {Status::RedisExecErr, s.ToString()};
+
+      *output = redis::Integer(cnt);
+      return Status::OK();
+    }
+};
+
 REDIS_REGISTER_COMMANDS(MakeCmdAttr<CommandJsonSet>("json.set", -3, "write", 1, 1, 1),
-                        MakeCmdAttr<CommandJsonGet>("json.get", -2, "read-only", 1, 1, 1), );
+                        MakeCmdAttr<CommandJsonGet>("json.get", -2, "read-only", 1, 1, 1),
+                        MakeCmdAttr<CommandJsonStrAppend>("json.strappend", -2, "read-only", 1, 1, 1), );
```

Comment:
I don't know why arity is -2. The syntax for it is:

```
JSON.STRAPPEND key [path] value
```

Seems that the arity should be `-3` since it should have at lease `JSON.STRAPPEND, key, value` ?

Reply:
should be -4, at lease `JSON.STRAPPEND, key, path, value`, should have one `path`

Reply:
From the RedisJson code impl[1] and document[3], I think it's -3

1. https://redis.io/commands/json.strappend/
2. https://github.com/RedisJSON/RedisJSON/blob/d4ba815fb13412c879b2cd3cd29ab60aee3c211b/redis_json/src/commands.rs#L939

But I think you can also use `4`.

Reply:
ok, thanks